Actions/ReadSettings/ReadSettings.ps1

```diff
@@ -139,13 +141,24 @@ try {
                     $_ -like $getEnvironments -and $_ -notlike '* (PROD)' -and $_ -notlike '* (Production)' -and $_ -notlike '* (FAT)' -and $_ -notlike '* (Final Acceptance Test)'
                 }
             } | Where-Object {
-                Write-Host "Environment: $_"
-                if ($ghEnvironments -contains $_) {
-                    # Environment is GitHub Environment, default branches are controlled on GitHub
-                    $branches = @( '*' )
+                $envName = $_
+                Write-Host "Environment: $envName"
+                $ghEnvironment = $ghEnvironments | Where-Object { $_.name -eq $envName }
+                if ($ghEnvironment) {
+                    $branchPolicy = ($ghEnvironment.protection_rules | Where-Object { $_.type -eq "branch_policy" })
+                    if ($branchPolicy) {
+                        Write-Host "GitHub Environment $envName has branch policies, getting branches from GitHub API"
+                        $branchesUrl = "$($ENV:GITHUB_API_URL)/repos/$($ENV:GITHUB_REPOSITORY)/environments/$([Uri]::EscapeDataString($envName))/deployment-branch-policies"
+                        Write-Host "Getting branches for $envName from GitHub API"
+                        $branches = @((InvokeWebRequest -Headers $headers -Uri $branchesUrl -ignoreErrors | ConvertFrom-Json).branch_policies | ForEach-Object { $_.name })
+                    }
+                    else {
+                        Write-Host "GitHub Environment $envName dot not have branch policies, using main as default"
```

Comment:
```suggestion
                        Write-Host "GitHub Environment $envName does not have branch policies, using main as default"
```

Reply:
Also, some legacy repos might still be using _master_ as default branch.

There's GitHub API to check the default branch: https://docs.github.com/en/rest/repos/repos?apiVersion=2022-11-28#get-a-repository

Reply:
AL-Go currently only supports main as default branch - there is a work-item to change this.
For AL-Go repos - master was never used.